### DISCUSSION TITLE
[product] Categorizing Expenses for Accounting

### DISCUSSION THREAD

**Discussion by iamronen** [opencollective/opencollective#5698] [Ideas]
This discussion thread is a conversation around a set of proposals for improving the way we handle categorization of expenses on the platform. This is currently based on a somewhat ambiguous implementation and application of tags which are used by different people, in different contexts in different ways.

**This proposal originated with a need identified within OSC to make it possible for collectives (open-source projects) to categorize their expenses in order to effectively communicate to their funders how money is used in the project.** 

The proposal has been posted as a markdown file in the opencollective repository:
https://github.com/opencollective/opencollective/blob/main/product/pitches/Categorizing%20%20Expenses%20for%20Accounting.md

The conversation may lead to changes in the original proposal and may lead to other proposals altogether.

**Comment by BenJam** [comment]
Love this, thanks for spending the time. Looking at this purely from a selfish, OSC perspective I want to reiterate the opening motivation:"

> [To] provide funders with more information about the impact of their investments

> [For collective admins] to be able to organize their expenses on the platform into [..] categories

The unexpected discoveries contain a lot of motivations that we should continue to explore (and I have questions about some of the findings there) but I want, for the moment to focus on the strong, stategically aligned motivations above. 

## Delivering Immediate Value

My biggest concern about the scopes presented is that there is a considerable investment from OC Inc. before we deliver value to the two users identified above. And that I beleive we have some of the tools avaiaible for us today to deliver some of that value cheaply and immediately (within one cycle). 

Why do that?

In my opinion we gain a lot by engaging users early. We gain feedback based on real world usage, we gian an ally to help us reach others and we gain a partner to co-design what comes next. My question with the proposal as written is 'where is the carrot?' Where do I as a collective admin, benefit from using these categories? For me we lost that when we excluded the visualisation from the initial scope and I think that is the key to creating that value for the user and unlocking the product development process for us as a team. 

## Proving the case

I want to stress that, for me, none of this proposal works without collective admins actively using these features. So my proposal is to deliver immediate value to them, and to judge our continued investment in this areas on our ability to engage them early on. 

For me, the success of an MVP would be judged solely on getting say, 10 collectives to categorise their expenses.

## An MVP using the tools we have today

We have a tool that _we_ (`/opensource`, `/foundation`, `/opencollective`) use to categorise expenses, **tags** the question is:

> Should we explicitly repurpose tags on expenses as categories of expenditure?

We can answer this question by looking at how tags are used outside of our own collectives. If we find that collective are already doing this then we should lean into that are create more intentionality around it, if they are being used for something else we may find that a greenfield approach of creating a new 'category' field on expenses is less work for Collective admins. 

### What are tags used for today?

[2,093 Collectives are using tags today, with 1,674 tags used]([url](https://opencollective-metabase.herokuapp.com/dashboard/114-tags)). Of those the biggest users are 

* `/transaidcymru` (684 of 775 expenses tagged, 197 tags)
* `/webpack` (575 of 769 expenses tagged, 19 tags)
* `/agora` (565 of 598 expenses tagged, 29 tags)
* `/wwcodeportland` (507 of 555 expenses tagged, 19 tags)
* `/wwcode` (507 of 545 expenses tagged, 14 tags)

Looking at some of the above we can see each of these collectives are using tags to categorise their expenditure. We see also that this proves out in general by looking at the most popular tags used:

* {"food & beverage"}	2,567
* {engineering}	2,384
* {communications}	2,214
* {"web services"}	2,081
* {travel}	1,052

### A freebie!

One thing to note is that we already have a nice categorisation for expenses **receipts** and **invoices** merely shoing a visualisation of expenditure on these two categories would demonstrate to admins and fundsers how much the collective is spending on people's time vs. reimbursing expenses. 

### Showing the value 

I think, given the above, there's a strong case for using tags to categorise expenses, and to focus inittially on showing this data on Collective profile pages to demonstrate the value of doing so to collective admins. I think the visualisations presented by @iamronen are great and we can see that projects like ESLint are already learning into them. I'd love @memo and @amina to look at this a little more. 

### Issues, Blockers and Detractors

The next question is, what are the problems associated with tags to categorise expensesthat we might need to address immediately?

**What's in a name?** For me, the first is possibly the name. Tags can be used by anyone everywheree for everything, they do not signla to the user their proposed intent. I think we may need to create that intent by calling tags 'categories' on expenses. 

**Who are categories for?** For me, categories are for admins to organise their Collective's expenditure. They are not for expense submitters to organise their receipts. Today a submitted can add a tag (category) to an expenses, we should stop this or at least provide an option. We should certianly not allow expense submitters to reclassify (tag) expense sonce submitted. 

**Standardisation** As is expense tags are scoped to the current collective. When adding a new tag, the tags previously used are presented for quick selection. I like the freedom here but we can forsee a problem in the future when we need to aggregate data across many collectives. I **do not** think that we should limit expense categories however because there is strong alignment between the benefits and the work required from collectives once this requirement is developed i.e. the benefit (large funders can understand how money is being used across all the collectives they support) is aligned with the work required of the collectives that probably already recieve a lot of support from those companyies (and stand to gain more). 

**Double spending** Expense tags (categories) are stored in an array currenctly, this leads to a double-count of expenses when they have multiple tags in metabase. I **do not** consider this an issue as only we have access to metabasee as a work around but **it is a consideration*** to take into account when cerating visualisations using this data. We may need to use documentaiton and guidance here at MVP to mitigate this issue. 

## The Proposal

Wrapping this all up I think, ahead of any further work we should:

- add visualisations to the collective profile page showing how expenses are categorised using tags, and by type (invoice, reimbursement)
- engage with the five Collectives above, and ESLint to have them categorise their expenses in a way that works for them
- show these collective profiel pages to some large, financial contributors for feedback

If we get positive feedback, and we see collecives using tags like this then I think we should carry on investing in some of the challenges above and described by @iamronen.

**Reply by iamronen** [reply]
question to @alanna and @natehn in the context of OCF ops: 
which of the two following options do you prefer and why?
1. Prevent expense submitters from modifying expense tags after expenses have been approved and tagged by a collective admin.
2. Completely prevent expense submitters from tagging expenses.

**Reply by iamronen** [reply]
@BenJam I am realizing that transactions also need to be incorporated into these views:
1. Transactions from a collective are an expense.
2. Transactions to a collective are contributions.

Is that correct?

**Reply by BenJam** [reply]
that's true, but i'm not sure we are really focussing on income at this point. I think we should focus on the spending side for the moment. the 'estiamted annual budget' on the income side is problematic but it works, and I'm not sure how much value having a graph is to other funders or admins.

**Reply by iamronen** [reply]
we still need to account for them as an expense!

as for contributions:
1. my draft proposal (sent to you) does include contributions 
2. every time I encounter the "estimated annual budget" I scratch my head wondering how that number was conjured up and what does it tell me.
3. I do think that being able to see a funding trajectory over time and a discernment between recurring and one-time contributions does tell you about the healthy and sustainability trajectory of a collective.
4. I think this is a relatively cheaper visualization to create (no tagging involved, we already have the data handy).

**Reply by natehn** [reply]
@iamronen I prefer option 1 - I would want submitters to be able to tag their expenses when submitting/before approved and paid

**Reply by alanna** [reply]
I prefer neither - OCF wants its Collectives to have full flexibility about how they use expense tags. For OCF's own ops budget, we can enforce our policies manually and I don't think technical enforcement would be preferred. If you want to create some kind of technical enforcement, a Collective level setting could be an option, like "Who can edit expense tags? submitters and admins, or only admins".

**Reply by iamronen** [reply]
Thank you for the inputs. 
I have placed on hold the question of permissions to add/edit tags.
If in the future collectives do encounter any frictions around this we will tend to it.

**Comment by iamronen** [comment]
another example of things we can (should?) be doing on the collective profile pages with the data that we have:
https://opencollective.slack.com/archives/C0HSLRNVC/p1656466043624139?thread_ts=1656336508.813459&cid=C0HSLRNVC
![image](https://user-images.githubusercontent.com/8337819/176453903-611b4252-ca69-4044-86f7-964e6f5134dc.png)

**Reply by natehn** [reply]
A related issue I submitted: https://github.com/opencollective/opencollective/issues/5553

**Reply by iamronen** [reply]
@natehn please have a look at the scope presented in #5704 

Please do comment on the issue if you wish to suggest specific additions for clarifying the "estimated yearly budget" that is currently displayed.

**Comment by poohlaga** [comment]
After reading through this and Ben's feedback it feels like this could be two development efforts.
1. reports for collectives
2. updates to expenses for admin categorization

for #2, I have two suggestions that will make it easier for us to manage accounting reports easier to manage at the end of the month and the end of the year when we have to prepare for tax filing:

2.a. An expense can have multiple line items and currently the tag is on the expense level. It would be ideal to move the categorization down a level for expenses that have multiple receipts. For example, we get one invoice from our HR solution providers but the cost is inclusive of their service fee and a salary for the employee. For tax purposes, we would ideally want to break this out into two line items with two categories ‘salary’ and ‘services’.

2.b. it would be good to flag an accounting category for tax purposes. for example, in accounting software, there is usually an admin level that lets a user choose user-created accounting categories representing salary & wages. like an umbrella “tax category” that holds multiple accounting categories.
at the end of the year, any expenses that fall under the accounting categories in the ‘tax category’ are compiled when you generate the 1099 report which looks for any one person that got paid over $600 US dollars that year. Only expenses that are salary & wages should be considered for this report.

Would love to explore this need more with input from @natehn @alinamanko and @alanna in particular

**Comment by Betree** [comment]
> Looking at some of the above we can see each of these collectives are using tags to categorise their expenditure. We see also that this proves out in general by looking at the most popular tags used:
> * {"food & beverage"}	2,567
> * {engineering}	2,384
> * {communications}	2,214
> * {"web services"}	2,081
> * {travel}	1,052

**Fun fact:** these are the most popular tags because Open Collective used to enforce a limited set of tags. At the time, the feature was called "category" and you could only pick one. Sounds familiar :smile:? Here's how [it looked](https://web.archive.org/web/20180506094246/https://opencollective.com/webpack/expenses):

![image](https://user-images.githubusercontent.com/1556356/181788367-84c6f46a-a64b-48db-80ba-2e63fb7e6fa3.png)

We moved to tags as part of https://github.com/opencollective/opencollective/issues/3438. I don't have the full context in mind (_if anyone wants to dig, the public meeting folder probably contains some discussions about that around `2019-11-01`_), but as I remember it:
- People wanted to have their own "categories" to finely represent the kind of expenses that they have
- People wanted to be able to add multiple tags because sometimes you submit an expense that falls under multiple "categories" (e.g. reimbursements for my travel expenses can fall into travel + food & beverage + accomodation). We did not have this issue before the expense flow redesign because you could only have one item per expense at the time.

I agree with @BenJam's proposal above to find ways to reuse tags to solve our problems. The concept proposed in https://github.com/opencollective/opencollective/blob/main/product/pitches/Categorizing%20%20Expenses%20for%20Accounting.md looks good, but it's a super complex project that is difficult to test with an MVP.

Re-using tags could have an MVP as simple as a rule saying that tags starting with `@` are mutually exclusive. So you could end up with an expense tagged like that: `@engineering #bounty #security #contributor`.

![image](https://user-images.githubusercontent.com/1556356/181795055-10ce10ea-fae6-4cff-840d-2ad1b6200390.png)

**Reply by iamronen** [reply]
Thank you for the additional context @Betree :) It is interesting to see how the wheels turn over time.

Issue #5704 is an integration of the wide-scope concept and @BenJam 's comments. 

As for the @ solution, it is technically correct but does not offer a good user experience. I think we will end up with double tags "engineering" and "@engineering", 

I would like to use a clearer UI pattern to indicate the "selected" category such as:
 - Highlight the first tag and allow replacing it by drag&dropping another tag to be first.
 - Treat the list of tags like a group of select-options where only one can be selected (this will require numerous visual queues).
